lib/rugged_adapter/git_layer_rugged.rb

```diff
@@ -188,13 +188,29 @@ def cat_file(options, sha)
         @repo.lookup(sha).read_raw
       end
 
-      def apply_patch(head_sha = 'HEAD', patch=nil)
-        false # Rewrite gollum-lib's revert so that it doesn't require a direct equivalent of Grit's apply_patch
-      end
-
-      def revert(path, sha1, sha2, ref)
-        # FIXME: See https://github.com/gollum/grit_adapter/pull/14
-        fail NotImplementedError
+      def revert(path, sha1, sha2)
+        options = {}
+        options[:paths] = [path] if path
+        patch = path ?
+          @repo.diff(sha2, sha1, options).first.diff : @repo.diff(sha2, sha1)
```

Comment:
I tend towards the convention to only use the ternary operator on a single line. Otherwise a normal if-else might be better.